klippy/extras/bed_mesh.py

```diff
@@ -415,6 +444,80 @@ def print_probed_positions(self, print_func):
             print_func(msg)
         else:
             print_func("bed_mesh: bed has not been probed")
+
+
+    def tilt_probe_finalize(self, offsets, positions):
+        x_offset = offsets[0]
+        y_offset = offsets[1]
+        z_offset = offsets[2]
+        self.gcode.respond_info \
+                ("offsets: %f %f %f" % (x_offset,y_offset,z_offset) );
+
+        params=self.bedmesh.z_mesh.probe_params
+
+        if self.relative_reference_index is not None:
+            # zero out probe z offset and
+            # set offset relative to reference index
+            z_offset = positions[self.relative_reference_index][2]
```

Comment:
Since you only have 3 points it is likely that this will raise an exception, and if it works I don't think it would give the desired result.  The relative_reference_index is applied to the mesh probed points.

I need to do a more in depth review of the math, but it seems to me that it would be sufficient to reference the z_offset to the first z value in `positions`, as the height of the plane shouldn't really matter.

Reply:
I didn't go into relative index and internals of the probbing, just copy-pasted there.
if self.relative_reference_index non none by default? it didn't complain, and code is working as supposed. I tilted bed and checked )